tests/integration/test_overrides.py

```diff
@@ -0,0 +1,216 @@
+):
+    pex_info = PexInfo.from_pex(pex)
+    assert list(sorted(expected_overrides)) == list(sorted(pex_info.overridden))
+
+    dists_to_versions = defaultdict(list)
+    for dist in pex_info.distributions:
+        project_name_and_version = ProjectNameAndVersion.from_filename(dist)
+        dists_to_versions[project_name_and_version.canonicalized_project_name].append(
+            project_name_and_version.canonicalized_version
+        )
+
+    expected_project_names_to_versions = {
+        ProjectName(project_name): [Version(version) for version in sorted(versions)]
+        for project_name, versions in expected_overridden_dists.items()
+    }
+    assert expected_project_names_to_versions == {
+        project_name: sorted(versions, key=str)
+        for project_name, versions in dists_to_versions.items()
+        if project_name in expected_project_names_to_versions
+    }, pex_info.dump(indent=2)
+
+
+@pytest.fixture
+def complex_requests_override_pex(tmpdir):
+    # type: (Any) -> str
+
+    pass
+
+
+skip_unless_compatible_with_requests_2_31_0 = pytest.mark.skipif(
+    PY_VER < (3, 7), reason="The requests version tested requires Python `>=3.7`."
+)
+
+
+@skip_unless_compatible_with_requests_2_31_0
+def test_override(
+    tmpdir,  # type: Any
+    complex_requests_override_pex,  # type: str
+):
+    # type: (...) -> None
+
+    pex = os.path.join(str(tmpdir), "pex")
+    run_pex_command(
+        args=[
+            "--python",
+            sys.executable,
+            "--python",
+            ensure_python_interpreter(PY310 if PY_VER < (3, 10) else PY39),
+            "requests==2.31.0",
+            "--override",
+            "urllib3==1.21; python_version >= '3.10'",
+            "--override",
+            "urllib3==1.20; python_version < '3.10'",
+            "--override",
+            "idna<2.5",
+            "-o",
+            pex,
+        ]
+    ).assert_success()
+
+    dists = [
+        dist for dist in PEX(pex).resolve() if ProjectName("requests") == dist.metadata.project_name
+    ]
+    assert 1 == len(dists)
+    requests = dists[0]
+    request_deps_by_project_name = {dep.project_name: dep for dep in requests.requires()}
+    assert (
+        Requirement.parse("urllib3<3,>=1.21.1")
+        == request_deps_by_project_name[ProjectName("urllib3")]
+    ), (
+        "Our multiple override test requires requests normally have an urllib3 lower bound of "
+        "1.21.1."
+    )
+    assert (
+        Requirement.parse("idna<4,>=2.5") == request_deps_by_project_name[ProjectName("idna")]
+    ), "Our single override test requires requests normally have an idna lower bound of 2.5."
+
+    assert_overrides(
+        pex,
+        expected_overrides=[
+            'urllib3==1.21; python_version >= "3.10"',
+            'urllib3==1.20; python_version < "3.10"',
+            "idna<2.5",
+        ],
+        expected_overridden_dists={"urllib3": ["1.20", "1.21"], "idna": ["2.4"]},
+    )
```

Comment:
@cognifloyd you can do pretty crazy stuff now between `--exclude` and `--override`. Use the power wisely.

Reply:
Thank you! That turned out to be very elegant.